src/c++/perf_analyzer/genai-perf/genai_perf/llm_metrics.py

```diff
@@ -418,7 +418,18 @@ def _get_profile_metadata(self, data: dict) -> None:
             elif data["endpoint"] == "v1/completions":
                 self._response_format = ResponseFormat.OPENAI_COMPLETIONS
             else:
-                raise ValueError(f"Unknown OpenAI endpoint: {data['endpoint']}")
+                # TPA-66: add PA metadata to handle this case
+                # When endpoint field is either empty or custom endpoint, fall
+                # back to parsing the response to extract the response format.
+                request = data["experiments"][0]["requests"][0]
+                response = request["response_outputs"][0]["response"]
+                if "chat.completion" in response:
```

Comment:
Thats a nice work around and you tagged it with a ticket to complete later.
Nice work.